crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -29,11 +29,17 @@ pub use self::serde::*;
 pub use map_entities::*;
 
 use crate::{archetype::ArchetypeId, storage::SparseSetIndex};
-use std::{
-    convert::TryFrom,
-    fmt, mem,
-    sync::atomic::{AtomicI64, Ordering},
-};
+use std::{convert::TryFrom, fmt, mem, sync::atomic::Ordering};
+
+#[cfg(target_has_atomic = "64")]
+use std::sync::atomic::AtomicI64 as AtomicInt;
+#[cfg(target_has_atomic = "64")]
+type Int = i64;
+
+#[cfg(not(target_has_atomic = "64"))]
+use std::sync::atomic::AtomicI32 as AtomicInt;
```

Comment:
In https://github.com/bevyengine/bevy/pull/2827, we decided to use `AtomicUsize` for `WorldIds`, since that is guaranteed to be available if you can get 'normal' `std` compiling.
Would it make sense to use `AtomicIsize` here on platforms without `AtomicI64` support - that is, once we work out all the wrapping behaviours properly?

Reply:
I think `AtomicIsize` is fine to use as the fallback here, although like you mention the difference probably only matters for 16-bit platforms. If it might increase compatibility for such platforms then I suppose it might as well use `isize` rather than `i32` – although as you point out it may be moot if there are other blockers on such platforms.